src/Features/Core/Portable/PullMemberUp/MembersPuller.cs

```diff
@@ -0,0 +1,408 @@
+                return null;
+            }
+
+            return new SolutionChangeAction(
+                string.Format(FeaturesResources.Add_to_0, result.Destination),
+                cancellationToken => PullMembersUpAsync(document, result, cancellationToken));
+        }
+
+        /// <summary>
+        /// Return the changed solution if all changes in result are applied.
+        /// </summary>
+        /// <param name="result">Contains the members to pull up and all the fix operations</param>>
+        internal async Task<Solution> PullMembersUpAsync(
+            Document document,
+            PullMembersUpAnalysisResult result,
+            CancellationToken cancellationToken)
+        {
+            if (result.Destination.TypeKind == TypeKind.Interface)
+            {
+                return await PullMembersIntoInterfaceAsync(document, result, cancellationToken).ConfigureAwait(false);
+            }
+            else if (result.Destination.TypeKind == TypeKind.Class)
+            {
+                return await PullMembersIntoClassAsync(document, result, cancellationToken).ConfigureAwait(false);
+            }
+            else
+            {
+                throw ExceptionUtilities.UnexpectedValue(result.Destination);
+            }
+        }
+
+        private IMethodSymbol FilterGetterOrSetter(IMethodSymbol getterOrSetter)
+        {
+            // We are pulling a public property, it could have a public getter/setter but 
+            // the other getter/setter is not.
+            // In this scenario, only the public getter/setter
+            // will be add to the destination interface.
+            return getterOrSetter?.DeclaredAccessibility == Accessibility.Public ? getterOrSetter : null;
+        }
+
+        private IMethodSymbol CreatePublicGetterAndSetter(IMethodSymbol getterOrSetter)
+        {
+            // Create a public getter/setter since user is trying to pull a non-public property to an interface.
+            // If getterOrSetter is null, it means this property doesn't have a getter/setter, so just don't generate it.
+            return getterOrSetter == null
+                ? getterOrSetter
+                : CodeGenerationSymbolFactory.CreateMethodSymbol(getterOrSetter, accessibility: Accessibility.Public);
+        }
+
+        private async Task<Solution> PullMembersIntoInterfaceAsync(
+            Document document,
+            PullMembersUpAnalysisResult result,
+            CancellationToken cancellationToken)
+        {
+            var solution = document.Project.Solution;
+            var solutionEditor = new SolutionEditor(solution);
+            var codeGenerationService = document.Project.LanguageServices.GetRequiredService<ICodeGenerationService>();
+            var destinationSyntaxNode = await codeGenerationService.FindMostRelevantNameSpaceOrTypeDeclarationAsync(
+                document.Project.Solution, result.Destination, default, cancellationToken).ConfigureAwait(false);
```

Comment:
1. just pass in 'solution'.
2. named-arg for 'default' . it's not clear what that is supposed to be for.